.github/workflows/ci.yml

```diff
@@ -235,6 +252,9 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
 
+      - name: Remove Infection
+        run: composer remove --dev infection/infection --no-update
```

Comment:
question: 
why do we need that? we do not remove manually any other dev deps for build job (and deps are cached anyway)

Reply:
Because if we don't do it, it will fail some jobs on 7.4 ☺️. Since we need it locally too, I could add it dynamically in the CI only in Infection job. We can't add it to dev-tools too, because it requires PHPUnit and it is installed in regular Composer, and we don't want 2 of them 🙂.